lib/cast.js

```diff
@@ -268,8 +268,11 @@ module.exports = function cast(schema, obj, options, context) {
           options.strictQuery :
           'strict' in options ?
             options.strict :
-            'strict' in schema._userProvidedOptions ? schema._userProvidedOptions.strict :
-              schema.options.strictQuery;
+            'strictQuery' in schema._userProvidedOptions ?
```

Comment:
Nitpick: This seems to be growing so much that it's not easy to understand it at all, I think we should refactor this to an external function that builds the strictQuery value.